src/pages/index.mdx

```diff
@@ -12,17 +9,387 @@ export const disableEditGithub = true
+
+void main() {
+  final profiles = Nitric.bucket("profiles").allow([
+    BucketPermission.read,
+    BucketPermission.write,
+    BucketPermission.delete,
+  ]);
+}
+```
+
+</CodeGroup>
+
+Nitric collects everything your services request. When you deploy, the deployment plugin you choose creates the resources and services, then links them all together.
+
+<Note>
+  That might sound like magic, but it's more like a compiler for
+  Infrastructure-as-Code - you can read about it in the
+  [Concepts](/concepts/introduction) section.
+</Note>
+
+---
+
+## Deployment Plugins (Providers)
+
+Nitric is designed to be independent of any platform, so you can run your apps anywhere. You can deploy to AWS, Azure, GCP, your own Kubernetes cluster or a single server. You could even deploy to multiple clouds at once.
+
+Are the differences between a bucket on AWS and a bucket on Azure important to most apps? We don't think so. So why should the code be different?
+
+Nitric abstracts away API layer differences, so you can focus on your app. The part that makes that possible is a plugin, we call a **Provider**.
+
+<CodeGroup>
+
+```yaml {{ label:"nitric.prod.yaml", title: "AWS Pulumi" }}
+provider: nitric/aws@1.1.1
+region: us-east-1
+```
+
+```yaml {{ label:"nitric.prod.yaml", title: "Azure Pulumi" }}
+provider: nitric/azure@1.1.1
+region: East US
+```
+
+```yaml {{ label:"nitric.prod.yaml", title: "Google Pulumi" }}
+provider: nitric/gcp@1.1.1
+region: us-east1
+```
+
+```yaml {{ label:"nitric.prod.yaml", title: "AWS Terraform" }}
+provider: nitric/awstf@1.1.1
+region: us-east-1
+```
+
+```yaml {{ label:"nitric.prod.yaml", title: "Google Terraform" }}
+provider: nitric/gcptf@1.1.1
+region: us-east1
+```
+
+```yaml {{ label:"nitric.prod.yaml", title: "Custom" }}
+# Build your own provider to deploy anywhere you like.
+provider: custom/on-prem@1.0.0
+```
+
+</CodeGroup>
+
+We have several providers built-in with IaC from [Pulumi](https://www.pulumi.com/) or [Terraform](https://www.terraform.io/), but you can build your own with any tools you prefer and **deploy anywhere**.
+
+---
+
+## Projects
+
+Projects built with Nitric don't have many restrictions. You can use most languages, libraries, tools, clouds, services, mostly anything you like. But, you need to have a `nitric.yaml` file in the root of your project.
+
+```yaml {{ label:"nitric.yaml" }}
+name: example
+services:
+  - match: services/*.ts
+    start: npm run dev:services $SERVICE_PATH
+```
+
+Nitric uses this to find your services, then it turns each service into a container, runs them in deployment mode to match the resources you requested and gives the result to the Provider - which either generates IaC (like Terraform) or automatically deploys your app.
+
+So, a project structure might look something like this:
+
+<CodeGroup>
+
+```plaintext {{ title:"JavaScript" }}
+example/
+├── nitric.yaml
+├── services/
+│   └── orders.ts
+│   └── users.ts
+└── package.json
+```
```

Comment:
Suggestion: Split into separate code types with correct extensions for JavaScript.
```plaintext {{ title:"JavaScript" }}
example/
├── nitric.yaml
├── services/
│   └── orders.js
│   └── users.js
└── package.json
```

```plaintext {{ title:"TypeScript" }}
example/
├── nitric.yaml
├── services/
│   └── orders.ts
│   └── users.ts
└── package.json
```

Reply:
Just updated the extensions to js for now